system/Encryption/Encryption.php

```diff
@@ -132,10 +131,11 @@ public function __construct(BaseConfig $config = null)
 	public function initialize(BaseConfig $config = null)
 	{
 		// override config?
-		if (! empty($config))
+		if ($config)
 		{
-			$this->driver = $config->driver;
 			$this->key    = $config->key;
+			$this->driver = $config->driver;
+			$this->digest = $config->digest;
```

Comment:
probably add default value:

```suggestion
                      $this->digest = $config->digest ?? 'SHA512';
```